app/src/main/java/com/example/playlistmaker/search/view_model/SearchViewModel.kt

```diff
@@ -65,6 +69,19 @@ class SearchViewModel(
         }
     }
 
+    fun searchDebounce(searchText: String) {
+        if (lastSearchText == searchText) {
+            return
+        } else {
+            lastSearchText == searchText
+            searchJob?.cancel()
+            searchJob = viewModelScope.launch {
+                delay(SEARCH_DEBOUNCE_DELAY)
+                searchRequest(searchText)
+            }
+        }
```

Comment:
Если ты тут делаешь `return` в первой ветке, то ветку else можно не писать, так будет меньше нагрузка при чтении:
```suggestion
        if (lastSearchText == searchText) return
        lastSearchText == searchText
        searchJob?.cancel()
        searchJob = viewModelScope.launch {
            delay(SEARCH_DEBOUNCE_DELAY)
            searchRequest(searchText)
        }
```